sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -841,42 +841,40 @@ class Evaluator(
   def visitObjComp(e: ObjBody.ObjComp, sup: Val.Obj)(implicit scope: ValScope): Val.Obj = {
     val binds = e.preLocals ++ e.postLocals
     val compScope: ValScope = scope // .clearSuper
-
-    lazy val newSelf: Val.Obj = {
```

Comment:
Since `newSelf` was no longer referenced after other changes, I decided to remove it to simplify the code and to avoid the risk of future mis-use of it.